sycl/doc/GettingStartedXilinxFPGA.md

```diff
@@ -528,3 +527,205 @@ the insane! Hopefully...
+* `-DLIBCLC_TARGETS_TO_BUILD="nvptx64--;nvptx64--nvidiacl"` needed for CUDA support;
+* `-DSYCL_ENABLE_XPTI_TRACING=ON` adds useful debugging capabilities;
+* `-DBUILD_SHARED_LIBS=ON` to build speedup;
+* `-DLLVM_TABLEGEN="$SYCL_HOME/build-Release/bin/llvm-tblgen" -DCLANG_TABLEGEN="$SYCL_HOME/build-Release/bin/clang-tblgen"` to reuse the tablegen built with the release version and speedup builds. The release build must be done before configuring the Debug one;
+* `-DCMAKE_CXX_FLAGS_DEBUG="-g -fstandalone-debug"` to have full debug information.
+
+* For details about the CMake configuration see https://llvm.org/docs/CMake.html
+
+while building the sycl-toolchain target, the device runtime for spirv and nvptx targets gets compiled using the device compiler from the specific build.
+building anything with a debug compiler is very slow, but it can be speedup using:
+
+```bash
+export LD_LIBRARY_PATH=$SYCL_HOME/build-Release/lib:$LD_LIBRARY_PATH
+```
+
+this will make the debug compiler select the dynamic libraries of the release compiler and speedup the build. This only works because the Release and Debug build have ABI compatible configuration, changing LLVM_ENABLE_ASSERTIONS or other configuration may change that.
+
+### Debugging
+
+#### Environnement variables
+
+some environment variables are very useful for debugging:
+
+```bash
+# Redirect the directory used as temporary directory 
+# for the compiler and various tools
+export TMP=$SYCL_HOME/../tmp
+
+# SYCL_PI_TRACE should always be at least at 1, this make the SYCL runtime emit logs about which device is selected
+export SYCL_PI_TRACE=1
+
+# SYCL_PI_TRACE can be set at -1 to have full debug information but this is quite noisy
+export SYCL_PI_TRACE=-1
+```
+
+#### Clang flags
+
+Some useful Clang flags:
+* `-ccc-print-phases` outputs the graph of compilation phases;
+* `-ccc-print-bindings` outputs the mapping from compilation phases to commands.
+* `-###` outputs the commands to run to compile. The driver will create files to reserve them for those commands.
+
+#### Running a single test
+
+To run test of the test suite in isolation use:
+```bash
+python3 /path/to/build/dir/bin/llvm-lit -v --param XOCC=all path/to/test.cpp
+```
+where all tests utilities must have been build for this to work.
+
+#### v++ Logs
+The kinds of following errors are typical of a back-end issue:
+```
+ERROR: [v++ 60-300] Failed to build kernel(ip) kernel_name, see log for details: ROOT/_x/sycl-xocc.ppArYN_kernel_name/kernel_name/vitis_hls.log
+ERROR: [v++ 60-599] Kernel compilation failed to complete
+ERROR: [v++ 60-592] Failed to finish compilation
+```
+the path contains a hash and kernel names so they wont match exactly.
+
+Sometimes the specified log file contains useful information
+but the more detailed log file `ROOT/_x/sycl-xocc.ppArYN_kernel_name/kernel_name/kernel_name/solution/.autopilot/db/autopilot.flow.log`
```

Comment:
Fix the cohesion of the sentence. I do not even if you are talking about 1 or 2 different log files here...

Reply:
Then?